src/main/java/org/mskcc/cmo/metadb/service/impl/ValidRequestCheckerImpl.java

```diff
@@ -271,9 +271,10 @@ private Boolean hasCmoPatientId(Map<String, String> sampleMap) {
      */
     private Boolean hasValidSpecimenType(Map<String, String> sampleMap) {
         //this can also be sampleClass
-        String specimenType = ObjectUtils.firstNonNull(String.valueOf(sampleMap.get("specimenType")),
-                String.valueOf(sampleMap.get("sampleClass")));
-        if (Strings.isBlank(specimenType)
+        String specimenType = String.valueOf(ObjectUtils.firstNonNull(sampleMap.get("specimenType"),
+               sampleMap.get("sampleClass")));
+        if (specimenType == "null" 
+                || Strings.isBlank(specimenType)
```

Comment:
Let's simplify and just check first if specimenType or sampleClass is present instead of handling a potential `"null"` 

```
String specimenType = sampleMap.containsKey("specimenType") ? sampleMap.get("specimenType") : sampleMap.get("sampleClass")
```